src/server.c

```diff
@@ -2725,10 +2725,8 @@ void homekit_server_on_pairings(client_context_t *context, const byte *data, siz
 
                     if (!pairing) {
                         // No admins left, enable pairing again
-                        INFO("Last admin pairing was removed, enabling pair setup");
-
-                        context->server->paired = false;
-                        homekit_setup_mdns(context->server);
+                        INFO("Last admin pairing was removed, resetting accessory");
+                        homekit_server_on_reset(context);
```

Comment:
I'm not sure about this one. I would suggest instead using an event callback and do reset if there are no pairings left.

Reply:
I am not sure about how to do that.